_modules/mac_kext.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import absolute_import, unicode_literals, print_function
+
+# Import python libs
+import logging
+import os
+
+# Import salt libs
+import salt.utils.platform
+from salt.exceptions import CommandExecutionError
+from salt.utils.versions import LooseVersion as _LooseVersion
+
+# Define the module's virtual name
+__virtualname__ = 'kext'
+
+def __virtual__():
+    '''
+    Only for macOS with launchctl
+    '''
+    if not salt.utils.platform.is_darwin():
+        return (False, 'Failed to load the mac_kext module:\n'
+                       'Only available on macOS systems.')
+
+    if not os.path.exists('/sbin/kextunload'):
+        return (False, 'Failed to load the mac_kext module:\n'
+                       'Required binary not found: "/sbin/kextunload"')
+
+    if not os.path.exists('/usr/sbin/kextstat'):
+        return (False, 'Failed to load the mac_kext module:\n'
+                       'Required binary not found: "/usr/sbin/kextstat"')
+
+    return __virtualname__
+
+
+log = logging.getLogger(__name__)
+
+
+def unload(name):
+    '''
+    Unload a kext at the given path.
+
+    name Path to Kext to unload.
+
+    '''
+    cmd = '/sbin/kextunload {}'.format(name)
+    ret = __salt__['cmd.run_all'](cmd)
+    # Raise an error or return successful result
+    if ret['retcode']:
+        out = 'Failed to {0} service:\n'.format(cmd)
+        out += 'stdout: {0}\n'.format(ret['stdout'])
```

Comment:
This pattern of catting the stdout, stderr, and retcode together is repeated a bunch of times. I would factor that into a helper function.

Also, not a huge deal, but probably a more idiomatic/performant version would be
```
out = (
    'Failed to {} service:\n'
    'stdout: {}\n'
    'stderr: {}\n'
    'retcode: {}').format(cmd, ret['stdout'], ret['stderr'], ret['retcode'])
```
I mean, wrap stuff however you find most beautiful. But rather than format each line and += building the string, just do it in one go.

When you do `str.format`, you can omit the number in the curly braces unless you're need to duplicate the same entry multiple times, e.g.
```
a = 'tacos'
b = 'burritos'
c = '{0} are better than {1} according to some sources, but others believe {1} are better than {0}'.format(a, b)
```